### PR TITLE
Feature/scheduled events endpoint

### PR DESCRIPTION
**Pull request by yevhen-turovets**
The task in Trello https://trello.com/c/fKE4ujXu

### CODE REVIEW

File: client/src/components/scheduled-events/Event.vue [hunk 1]
```diff
@@ -1,21 +1,26 @@
 <template>
     <div class="event">
         <VContainer class="event-date">
-            <span>{{ scheduledEvent.date }}</span>
+            <span>
+                {{ this.getEventDate(scheduledEvent.startDate) }}
```
Comment (dmitriy-uvin) [inline client/src/components/scheduled-events/Event.vue:5]: just `getEventDate(scheduledEvent.startDate)`

File: client/src/components/scheduled-events/Event.vue [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
+                                :color="scheduledEvent.eventType.color"
                             />
+
                             <span class="time">
-                                {{ scheduledEvent.time }}
+                                {{
+                                    this.getDurationTime(
+                                        scheduledEvent.startDate,
+                                        scheduledEvent.eventType.duration
+                                    )
```
Comment (dmitriy-uvin) [inline client/src/components/scheduled-events/Event.vue:22]: just `getDurationTime(scheduledEvent.startDate, scheduledEvent.eventType.duration)`

File: client/src/components/scheduled-events/Event.vue [hunk 3]
```diff
@@ -158,6 +202,7 @@ export default {
             line-height: 20px;
             letter-spacing: 0.25px;
             color: #2c2c2c;
```
Comment (dmitriy-uvin) [inline client/src/components/scheduled-events/Event.vue:204]: check if you can yse variable from the theme

Comment (yevhen-turovets) [reply client/src/components/scheduled-events/Event.vue:204]: I check this when created layout - don't have variables for this colors

**Comment by dmitriy-uvin** [comment]
Fix errors.
Frontend approved.

File: api/app/Actions/Event/GetEventCollectionAction.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    {
+        $this->repository = $eventRepository;
+    }
+
+    public function execute(GetEventCollectionRequest $request): PaginatedResponse
+    {
+        $criteria = [new OwnerCriterion(Auth::id())];
+
+        if ($request->getStartDate()) {
+            $startDate = Carbon::parse($request->getStartDate())->format('Y-m-d');
```
Comment (aLLeXUs) [inline api/app/Actions/Event/GetEventCollectionAction.php:31]: `start_date` and `end_date` should contain time too for showing correct results for past and upcoming events.

File: api/app/Http/Presenters/EventPresenter.php [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        return [
+            'id' => $event->id,
+            'invitee_name' => $event->invitee_name,
+            'invitee_email' => $event->invitee_email,
+            'invitee_question' => 'No questions',
+            'start_date' => $event->start_date,
+            'location' => $event->location,
+            'timezone' => $event->timezone,
+            'created_at' => Carbon::parse($event->created_at)->format('d.m.Y'),
```
Comment (aLLeXUs) [inline api/app/Http/Presenters/EventPresenter.php:32]: The date and time should be passed. You will be able to convert it to the needed format on the client.

File: api/database/factories/UserFactory.php [hunk 6]
```diff
@@ -19,7 +19,7 @@
 */
 
 $factory->define(User::class, function (Faker $faker) {
-    $email = $faker->unique()->safeEmail;
+    $email = "schedulia-test@schedulia.xyz";
```
Comment (aLLeXUs) [inline api/database/factories/UserFactory.php]: I am not sure we should hardcode only one email here. We should be able to seed N amount of users.

File: api/database/migrations/2020_08_27_104237_add_location_to_events_table.php [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('events', function (Blueprint $table) {
+            $table->string('location')->nullable();
```
Comment (aLLeXUs) [inline api/database/migrations/2020_08_27_104237_add_location_to_events_table.php:17]: Probably we will need one more field - `location_type` (ex. address, zoom etc.). `location` can be used as the data for `location_type` (ex. place name or coordinates for address, URL for zoom etc.)

**Review by aLLeXUs — CHANGES_REQUESTED** [review]

File: api/database/migrations/2020_08_27_104237_add_location_to_events_table.php (hunk 7 above)
Comment (yevhen-turovets) [reply api/database/migrations/2020_08_27_104237_add_location_to_events_table.php:17]: do I need to add a location type field?

Comment (aLLeXUs) [reply api/database/migrations/2020_08_27_104237_add_location_to_events_table.php:17]: ok, you can skip it for now

**Review by aLLeXUs — APPROVED** [review]

File: client/src/store/modules/scheduledEvent/normalizer.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { eventTypeMapper } from '../eventType/normalizer';
+
+export const eventMapper = Event => ({
+    id: Event.id,
+    name: Event.invitee_name,
+    email: Event.invitee_email,
+    question: Event.invitee_question,
+    startDate: Event.start_date,
+    startTime: '',
```
Comment (aLLeXUs) [inline client/src/store/modules/scheduledEvent/normalizer.js:9]: No longer needed

**Pull request merged**